app/services/users.ts

```diff
@@ -0,0 +1,16 @@
+import { EntityRepository, Repository, getCustomRepository } from 'typeorm';
+import { User } from '../models/user';
+
+// eslint-disable-next-line new-cap
+@EntityRepository(User)
+class UserRepository extends Repository<User> {
+  public findByName(username: string): Promise<User | undefined> {
+    return this.findOne({ username });
+  }
+
+  public createAndSave(user: User): Promise<User> {
+    return this.save(user);
+  }
+}
+
+export default (): UserRepository => getCustomRepository(UserRepository);
```

Comment:
Im not very sure if this approach is the best option. I would consider something like this...

```javascript
import { getRepository } from 'typeorm';
import { User } from './entity/User';

const userRepository = getRepository(User);

export function findByName(username: string): Promise<User | undefined> {
  return userRepository.findOne({ username });
}

export function createAndSave(user: User): Promise<User> {
  return userRepository.save(user);
}

export default {
  findByName,
  createAndSave
}
```

Reply:
I like this option, but `getRepository` is not possible like this, because need a connection to database and when service is loaded no connection is established yet

Reply:
It seems like repositories are internally something like singletons, check [this](https://github.com/typeorm/typeorm/issues/3879#issuecomment-479689032). Call as a function each time seems fine and is not creating a new one all time